UDPSocket.cpp

```diff
@@ -344,13 +346,13 @@ bool CUDPSocket::write(const unsigned char* buffer, unsigned int length, const s
 
 void CUDPSocket::close()
 {
-	for (int i = 0; i < UDP_SOCKET_MAX; i++)
-		close(m_fd[i]);
+	for (unsigned int i = 0; i < UDP_SOCKET_MAX; i++)
+		close(i);
```

Comment:
key to to the fix

Reply:
Yes, instead if passing file descriptor as an array index (and corrupt memory, like timers in DMRGateay's CDMRNetwork class, messing the timeout values).